firmware/console/binary/tunerstudio.cpp

```diff
@@ -322,38 +321,38 @@ void handleWriteChunkCommand(ts_channel_s *tsChannel, ts_response_format_e mode,
 		void *content) {
 	tsState.writeChunkCommandCounter++;
 
-	scheduleMsg(&tsLogger, "WRITE CHUNK mode=%d p=%d o=%d s=%d", mode, currentPageId, offset, count);
+	scheduleMsg(&tsLogger, "WRITE CHUNK mode=%d o=%d s=%d", mode, offset, count);
 
-	if (offset > getTunerStudioPageSize(currentPageId)) {
+	if (offset >= getTunerStudioPageSize()) {
 		scheduleMsg(&tsLogger, "ERROR invalid offset %d", offset);
 		tunerStudioError("ERROR: out of range");
 		offset = 0;
 	}
 
-	if (count > getTunerStudioPageSize(currentPageId)) {
+	if (count >= getTunerStudioPageSize()) {
```

Comment:
i wonder about the change here, could it be a bug? in case of offset 0?

good thing we do not have such big packets. also maybe we shall compare offset+count, not just count?

Reply:
We now do a full bounds check for all read/write operations!